src/main/java/org/jenkinsci/plugins/envinject/EnvInjectEnvVarsContributor.java

```diff
@@ -0,0 +1,27 @@
+package org.jenkinsci.plugins.envinject;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.model.*;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+@Extension
+public class EnvInjectEnvVarsContributor extends EnvironmentContributor {
+
+    @Override
+    public void buildEnvironmentFor(Job job, EnvVars env, TaskListener listener) throws IOException, InterruptedException {
+        for (Object property : job.getProperties().values()) {
```

Comment:
@oleg-nenashev I don't know how job.getProperty() can be `null` if this method is invoked only if `job.getProperties().values()` returns values.

Do you prefer to use `job.getProperty(EnvInjectJobProperty.class)`?